eclass/zig.eclass

```diff
@@ -0,0 +1,565 @@
+		ZBS_ARGS_BASE+=(
+			--search-prefix "${ESYSROOT}/usr/"
+			--sysroot "${ESYSROOT}/"
+		)
+	fi
+}
+
+# @FUNCTION: zig_pkg_setup
+# @DESCRIPTION:
+# Sets up environmental variables for Zig toolchain
+# and basic args for Zig Build System.
+zig_pkg_setup() {
+	[[ "${MERGE_TYPE}" != binary ]] || return 0
+
+	zig-utils_setup
+	zig_init_base_args
+
+	mkdir "${T}/zig-cache/" || die
+
+	# Environment variables set by this eclass.
+
+	# Used by Zig Build System to find `pkg-config`.
+	# UPSTREAM Used only by 9999 for now, should land in future
+	# 0.14 release.
+	export PKG_CONFIG="${PKG_CONFIG:-"$(tc-getPKG_CONFIG)"}"
+	# Used by whole Zig toolchain (most of the sub-commands)
+	# to find local and global cache directories.
+	export ZIG_LOCAL_CACHE_DIR="${T}/zig-cache/local/"
+	export ZIG_GLOBAL_CACHE_DIR="${T}/zig-cache/global/"
+}
+
+# @FUNCTION: zig_live_fetch
+# @USAGE: [<args>...]
+# @DESCRIPTION:
+# Fetches packages, if they exist, to the "ZBS_ECLASS_DIR/p/".
+# If you have some lazy dependency which is not triggered in default
+# configuration, pass options like you would pass them for regular
+# "ezig build".  Try to cover all of them before "src_configure".
+# **Note**: this function will be deprecated once/if
+# https://github.com/ziglang/zig/pull/19975 lands.
+#
+# Example:
+# @CODE
+# src_unpack() {
+# 	# If there are no lazy dependency:
+# 	zig_live_fetch
+#
+# 	# If there are lazy dependencies that can be triggered together:
+# 	zig_live_fetch -Denable-wayland -Denable-xwayland
+#
+# 	# If there are 2 lazy dependencies that can't be triggered
+# 	# together in one call because they conflict:
+# 	zig_live_fetch -Dmain-backend=opengl
+# 	zig_live_fetch -Dmain-backend=vulkan
+# }
+# @CODE
+zig_live_fetch() {
+	# This function will likely be called in src_unpack,
+	# before [zig_]src_prepare, so this directory might not
+	# exist yet.
+	mkdir -p "${BUILD_DIR}" > /dev/null || die
+	pushd "${BUILD_DIR}" > /dev/null || die
+
+	local args=(
+		"${ZBS_ARGS_BASE[@]}"
+
+		--global-cache-dir "${ZBS_ECLASS_DIR}/"
+
+		# Function arguments
+		"${@}"
+	)
+
+	einfo "ZBS: live-fetching with:"
+	einfo "${args[@]}"
+	ezig build --help "${args[@]}" > /dev/null
+
+	popd > /dev/null || die
+}
+
+# @FUNCTION: zig_src_unpack
+# @DESCRIPTION:
+# Unpacks every archive in SRC_URI and ZBS_DEPENDENCIES,
+# in that order.
+zig_src_unpack() {
+	# Thanks to Alfred Wingate "parona" for inspiration here:
+	# https://gitlab.com/Parona/parona-overlay/-/blob/874dcfe03116574a33ed51f469cc993e98db1fa2/eclass/zig.eclass
+
+	if [[ "${#ZBS_DEPENDENCIES_SRC_URI}" -eq 0 ]]; then
+		default_src_unpack
+		return
+	fi
+
+	local zig_deps=()
+	for dependency in "${!ZBS_DEPENDENCIES[@]}"; do
+		zig_deps+=("${dependency}")
+	done
+
+	# First unpack non-Zig dependencies, so that
+	# tarball with all Git dependencies tarballs is unpacked early.
+	local dist
+	for dist in ${A}; do
+		if has "${dist}" "${zig_deps[@]}"; then
+			continue
+		fi
+		unpack "${dist}"
```

Comment:
I think this would be easier to follow:

```suggestion
		if ! has "${dist}" "${zig_deps[@]}"; then
			unpack "${dist}"
		fi
```

Reply:
fda5525fd68b01b8548a8e1458b5bddfa9f6a7b4